plugins/thread.py

```diff
@@ -20,46 +21,43 @@ def __init__(self, wx):
 
     def run(self):
         # initializing
-        self.report(0)
+        self.progress(0)
 
-        temp_dir = tempfile.mkdtemp()
-        gerber_dir = os.path.join(temp_dir, 'gerber')
-        assembly_dir = os.path.join(temp_dir, 'assembly')
+        timestamp = datetime.datetime.now().strftime('%Y-%m-%d %H-%M-%S')
 
-        os.makedirs(gerber_dir)
-        os.makedirs(assembly_dir)
+        temp_dir = tempfile.mkdtemp() + timestamp
+        os.makedirs(temp_dir)
 
         _, temp_file = tempfile.mkstemp()
         project_directory = os.path.dirname(self.process_manager.board.GetFileName())
-        output_path = os.path.join(project_directory, outputFolder)
 
         try:
-            # configure and generate gerber
-            self.report(5)
-            self.process_manager.generate_gerber(gerber_dir)
+            # generate gerber
+            self.progress(5)
+            self.process_manager.generate_gerber(temp_dir)
 
             # generate drill file
-            self.report(15)
-            self.process_manager.generate_drills(gerber_dir)
+            self.progress(15)
+            self.process_manager.generate_drills(temp_dir)
 
             # generate netlist
-            self.report(25)
-            self.process_manager.generate_netlist(assembly_dir)
+            self.progress(25)
+            self.process_manager.generate_netlist(temp_dir)
 
             # generate pick and place file
-            self.report(40)
-            self.process_manager.generate_positions(assembly_dir)
+            self.progress(40)
+            self.process_manager.generate_positions(temp_dir)
 
             # generate BOM file
-            self.report(60)
-            self.process_manager.generate_bom(assembly_dir)
+            self.progress(60)
+            self.process_manager.generate_bom(temp_dir)
 
             # generate production archive
-            self.report(75)
-            temp_file = self.process_manager.generate_archive(gerber_dir, temp_file)
+            self.progress(75)
+            temp_file = self.process_manager.generate_archive(temp_dir, temp_file)
```

Comment:
Use shutil.make_archive(..., exclude_ext=['.csv', '.ipc'])

Reply:
shutil.make_archive() does not have a ignore or exclude parameter like shutil.copytree() does